### ISSUE TITLE
COMMAND PALLET: Shortcut keys are not read by screenreader

### ISSUE THREAD

**Issue by zersiax** [Microsoft/vscode#3839]
I just did a quick check on the command pallet and can navigate it just fine. I can read the actions, but it seems the shortcut keys assigned to these actions aren't read aloud by screenreaders. Obviously I can't verify if this is just me having a lot of commands with no keyboard shortcuts attached. Can anyone confirm?
Also, pressing escape dismisses the pallet but doesn't take your focus back to the editor. I think although focus stealing is usually frowned upon, in this case it would smooth out the interaction somewhat since you do hop to the command pallet by pressing f1, but have to use browse mode or tab + shift+tab to get back to where you were after it has been dismissed.

**Comment by bpasero** [comment]
@zersiax great catch! The ARIA label I am now putting with a keybinding is: "action label, shortcut theShortcut, commands". Maybe you have feedback if that is OK to do or not.

**Comment by bpasero** [comment]
(basically I am wondering if the shortcut should just follow the command label or if it should be prefixed with the word "shortcut").

**Comment by zersiax** [comment]
Either will work really. Adding shortcut does add some translation
overhead so make sure that is a localizable string if you go that
route, but I don't think it will be necessary since most shortcuts
will contain at least one of ctrl, alt, shift or windows which will
already denote it as a shortcut key. Am I overgeneralizing?

**Comment by bpasero** [comment]
@zersiax actually we seem to use real unicode characters to represent the keybinding. Would you think that is ok or should we find a readable label here?

One example is: ⇧⌘Y

**Comment by zersiax** [comment]
As it stands I don't think these unicode characters are currently
supported. I recommend using readable labels in this case, e.g. ctrl,
alt, Windows, End, not sure what the conventions are for things like
page-up and page-down.

**Comment by bpasero** [comment]
@zersiax ok makes sense

**Comment by zersiax** [comment]
Just tested your changes in this matter, they work well. I really
think though that adding 'shortcut'  in this label is superfluous, you
may leave it out and spare yourself the i18n hell it might cause :)
Also, the last bit of he label for each command reads ' commands,
picker' . Is there a reason for this?

**Comment by bpasero** [comment]
@zersiax ok I am removing the extra 'shortcut'. As for 'commands' being read, I thought it would be nice that you get to know in what context you are. The widget used for the commands list is very dynamic, it shows a lot of other things, not only commands. For example, we use ">" as prefix in the input field to show commands, once you remove ">" you end up in the list of files. If you type "?" you end up in a list of help entries that teach you how to use the widget with other prefixes.

I use this pattern in other places too, e.g. debug. When I use voice over I can see that voice over appends more contextual information with comma, for example "file.ts, files explorer, list box". I thought I would pick up this way of adding context to a label by adding my own context (the "List box" is added by voice over based on the ARIA role).

**Comment by zersiax** [comment]
ahh :) It makes sense then. I thought the commands pallet was just
that and in that case it would've been a little verbose. But I think
in this case it's just right :)

**Comment by bpasero** [comment]
Cool!
